hooks/playbooks/observability.yml

```diff
@@ -0,0 +1,56 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Deploy cluster observability operator.
+  gather_facts: false
+  hosts: "{{ cifmw_target_hook_host | default('localhost') }}"
+
+  tasks:
+    - name: Install cluster observability operator.
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: operators.coreos.com/v1alpha1
+          kind: Subscription
+          metadata:
+            name: observability-operator
+            namespace: openshift-operators
+            labels:
+              operators.coreos.com/observability-operator.openshift-operators: ""
+          spec:
+            channel: development
+            installPlanApproval: Automatic
+            name: cluster-observability-operator
+            source: redhat-operators
+            sourceNamespace: openshift-marketplace
+        validate_certs: false
+        api_key: "{{ cifmw_openshift_token | default(omit) }}"
+        host: "{{ cifmw_openshift_api | default(omit) }}"
+        username: "{{ cifmw_openshift_user | default(omit) }}"
+        password: "{{ cifmw_openshift_password | default(omit) }}"
+        kubeconfig: "{{ cifmw_openshift_kubeconfig | default(omit) }}"
+
+    - name: Wait until observability operator deployment is successful.
```

Comment:
I think that there is some time between the above Subscription is created and OpenShift creates the Deployment, as we had to implement custom waits in our CI: https://github.com/openstack-k8s-operators/telemetry-operator/blob/main/ci/create-coo-subscription.yaml#L28, but I never used the kubernetes.core collection so this might be already taking that in account.

Reply:
Thank you. I have include a check for the pod before waiting on deployment.